oscar/apps/dashboard/catalogue/views.py

```diff
@@ -115,13 +115,9 @@ def apply_search(self, queryset):
         data = self.form.cleaned_data
 
         if data.get('upc'):
-            # If there's an exact UPC match, it returns just the matched
-            # product. Otherwise does a broader icontains search.
-            qs_match = queryset.filter(upc=data['upc'])
-            if qs_match.exists():
-                queryset = qs_match
-            else:
-                queryset = queryset.filter(upc__icontains=data['upc'])
```

Comment:
Instead of your changes, what do you think of just doing the following?

``queryset = queryset.filter(Q(upc__icontains=data['upc']) | parent__upc__icontains=data['upc']))``

That way, we don't redefine the queryset and keep the ``exists`` logic. I also find it slightly more readable.